drivers/espi/espi_it8xxx2.c

```diff
@@ -1755,6 +1766,28 @@ void espi_it8xxx2_enable_pad_ctrl(const struct device *dev, bool enable)
 	}
 }
 
+void espi_it8xxx2_enable_trans_irq(const struct device *dev, bool enable)
+{
+	const struct espi_it8xxx2_config *const config = dev->config;
+
+	if (enable) {
+		irq_enable(IT8XXX2_TRANS_IRQ);
+	} else {
+		irq_disable(IT8XXX2_TRANS_IRQ);
+		/* Clear pending interrupt */
+		it8xxx2_wuc_clear_status(config->wuc.wucs, config->wuc.mask);
+	}
+}
+
+static void espi_it8xxx2_trans_isr(const struct device *dev)
+{
+	/*
+	 * This interrupt is only used to wake up CPU, there is no need to do
+	 * anything in the isr in addition to disable interrupt.
+	 */
+	espi_it8xxx2_enable_trans_irq(dev, false);
```

Comment:
I think this should clear the pending bit, but not disable the interrupt: that should be left to the wakeup code in `soc.c`.

If the IRQ gets disabled here, a wakeup can still be lost if this interrupt fires immediately after it gets enabled and before the chip goes to doze. If this ISR only clears the pending interrupt, then another transaction will still wake the CPU as desired.

Reply:
In soc.c, we enable this interrupt after disabling global interrupt (csr_clear(mie, MIP_MEIP);). So pending interrupt will not fire immediately until we enable global interrupt. (This is our hardware mechanism, a pennding interrupt is able to wake up CPU but dot not jump to isr if mie is disabled). Since we disable this interrupt before enabling global interrupt after wakeup, EC will not jump to this isr to save interrupt cost.

This disable is added in isr for safety, to make sure we won't hit interrupt storm if I missed something in the sequence.

Reply:
Okay, thanks for clarifying.